include/arbor/communication/mpi_error.hpp

```diff
@@ -0,0 +1,110 @@
+#pragma once
```

Comment:
This header is public, however it is only used in non-public code. And only in non-public cpp files at that.

Reply:
`mpi_error` should be able to be consumed though in user code, i.e. one should be able to write a `catch (mpi_error& e)` clause in an application.